src/bh/utils.js

```diff
@@ -57,3 +59,29 @@ export const formatters = {
   removeBreakLines,
   removeMultipleWhiteSpaces,
 };
+
+const exampleSearchFilterWithRegex = /((intent|label|entity):([a-zA-Z0-9_-]+))/g;
+
+const extractGroupsOfRegex = (regularExpression, value) => {
+  let match;
+  const regexGroups = {};
+  /* eslint-disable no-cond-assign */
+  while (match = regularExpression.exec(value)) {
+    Object.assign(regexGroups, { [match[2]]: match[3] });
```

Comment:
Make this function generic, just return a list of groups.
Do this parse in `exampleSearchToDicty`.